pkg/platform/runtime/setup/implementations/alternative/artifact.go

```diff
@@ -23,7 +23,23 @@ func NewArtifactSetup(artifactID artifact.ArtifactID, store *store.Store) *Artif
 	return &ArtifactSetup{artifactID, store}
 }
 
-func (as *ArtifactSetup) EnvDef(tmpDir string) (*envdef.EnvironmentDefinition, error) {
+func (as *ArtifactSetup) PrepareEnvDef(tmpDir, installDir string, constants envdef.Constants) (*envdef.EnvironmentDefinition, error) {
```

Comment:
Can you add a comment explaining the intend of this function? By the naming it sounds like this is some sort of staged function that you're only meant to call at a specific time.

Reply:
I agree the naming isn't perfect. This could still be named `EnvDef` and act as a constructor with added steps.